src/ui/src/app/Service/wolf-den.service.ts

```diff
@@ -4,20 +4,48 @@ import { Observable } from 'rxjs';
 import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
 import { IEmployeeDirectoryWithPagecount } from '../interface/iemployee-directory-with-pagecount';
 import { EmployeeService } from './employee.service';
+import { ToastrService } from 'ngx-toastr';
+import { Router } from '@angular/router';
 @Injectable({
   providedIn: 'root'
 })
 
 export class WolfDenService {
 
+  emp = inject(EmployeeService);
+  toastr=inject(ToastrService);
+  router=inject(Router)
 
-  private baseUrl=environment.apiUrl;
-  public userId : number=0;
-  public role:string="";
-  public firstName:string="Welcome Back";
+  private baseUrl = environment.apiUrl;
+  public userId: number = 0;
+  public role : string = "";
+  public firstName: string = ""; 
 
 
-  constructor(private http: HttpClient, private employeeService: EmployeeService) { }
+  constructor(private http: HttpClient, private employeeService: EmployeeService) {
+    if (localStorage.getItem('token') !== null) {
+      const payload = this.emp.decodeToken();
+      this.userId = parseInt(payload.EmployeeId || 0, 10);
```

Comment:
If you are using `EmployeeId` here, then rename the class member `userId` to `employeeId`.
In our system, userId signifies the user from Identity Table, to avoid ambiguity lets rename this variable

Reply:
done